modules/tinymce/src/plugins/image/main/ts/ui/Dialog.ts

```diff
@@ -353,36 +335,17 @@ const makeDialogBody = (info: ImageDialogInfo): DialogType.TabPanelSpec | Dialog
   }
 };
 
-const makeDialog = (helpers: Helpers) => (info: ImageDialogInfo): DialogType.DialogSpec<ImageDialogData> => {
-  const state = createState(info);
-  return {
-    title: 'Insert/Edit Image',
-    size: 'normal',
-    body: makeDialogBody(info),
-    buttons: [
-      {
-        type: 'cancel',
-        name: 'cancel',
-        text: 'Cancel'
-      },
-      {
-        type: 'submit',
-        name: 'save',
-        text: 'Save',
-        primary: true
-      }
-    ],
-    initialData: fromImageData(info.image),
-    onSubmit: helpers.onSubmit(info),
-    onChange: changeHandler(helpers, info, state),
-    onClose: closeHandler(state)
-  };
-};
-
-const submitHandler = (editor: Editor) => (info: ImageDialogInfo) => (api: API): void => {
+const submitHandler = (editor: Editor, info: ImageDialogInfo, helpers: Helpers) => (api: API): void => {
   const data: ImageDialogData = Merger.deepMerge(fromImageData(info.image), api.getData());
 
-  editor.execCommand('mceUpdateImage', false, toImageData(data, info.hasAccessibilityOptions));
+  // The data architecture relies on passing everything through the style field for validation.
+  // Since the style field was removed that process must be simulated on submit.
+  const finalData = changeStyle2(helpers.parseStyle, helpers.serializeStyle, {
```

Comment:
Could we rename this now that we have deleted the original `changeStyle` function?

Reply:
I thought about it, but at the time I was planning for this to be straight up reverted in the future. I now realise that isn't a viable solution to the long term problem so yeah I'll rename it.

Reply:
I hope we never put that input field back it has been a pain to handle updating that and the other fields since they need to be in sync.

Reply:
I think we will, but it will have validation so we can make it clear when things are wrong instead of simply blocking it.

Reply:
Turns out `changeStyle2` wasn't the important part of this code, it was the `style` attribute I was passing in to it. I deleted the method and all tests are still good.